core/bentley/src/StatusCategory.ts

```diff
@@ -0,0 +1,355 @@
+  export class Accepted extends SuccessCategory { public name = "Accepted"; public code = 202; }
+  export class NoContent extends SuccessCategory { public name = "NoContent"; public code = 204; }
+
+  export class BadRequest extends ErrorCategory { public name = "BadRequest"; public code = 400; }
+  export class Unauthorized extends ErrorCategory { public name = "Unauthorized"; public code = 401; }
+  export class Forbidden extends ErrorCategory { public name = "Forbidden"; public code = 403; }
+  export class NotFound extends ErrorCategory { public name = "NotFound"; public code = 404; }
+  export class MethodNotAllowed extends ErrorCategory { public name = "MethodNotAllowed"; public code = 405; }
+  export class NotAcceptable extends ErrorCategory { public name = "NotAcceptable"; public code = 406; }
+  export class RequestTimeout extends ErrorCategory { public name = "RequestTimeout"; public code = 408; }
+  export class Conflict extends ErrorCategory { public name = "Conflict"; public code = 409; }
+  export class Gone extends ErrorCategory { public name = "Gone"; public code = 410; }
+  export class PreconditionFailed extends ErrorCategory { public name = "PreconditionFailed"; public code = 412; }
+  export class ExpectationFailed extends ErrorCategory { public name = "ExpectationFailed"; public code = 417; }
+  export class MisdirectedRequest extends ErrorCategory { public name = "MisdirectedRequest"; public code = 421; }
+  export class UnprocessableEntity extends ErrorCategory { public name = "UnprocessableEntity"; public code = 422; }
+  export class Locked extends ErrorCategory { public name = "Locked"; public code = 423; }
+  export class FailedDependency extends ErrorCategory { public name = "FailedDependency"; public code = 424; }
+  export class UpgradeRequired extends ErrorCategory { public name = "UpgradeRequired"; public code = 426; }
+  export class PreconditionRequired extends ErrorCategory { public name = "PreconditionRequired"; public code = 428; }
+  export class TooManyRequests extends ErrorCategory { public name = "TooManyRequests"; public code = 429; }
+
+  export class InternalServerError extends ErrorCategory { public name = "InternalServerError"; public code = 500; }
+  export class NotImplemented extends ErrorCategory { public name = "NotImplemented"; public code = 501; }
+}
+
+class Success extends HTTP.OK { }
+
+class Pending extends HTTP.Accepted { }
+
+class NoContent extends HTTP.NoContent { }
+class NothingToDo extends HTTP.NoContent { }
+
+class BadRequest extends HTTP.BadRequest { }
+
+class Forbidden extends HTTP.Forbidden { }
+class PermissionsViolation extends HTTP.Forbidden { }
+class ReadOnly extends HTTP.Forbidden { }
+
+class NotFound extends HTTP.NotFound { }
+
+class NotEnabled extends HTTP.MethodNotAllowed { }
+class NotSupported extends HTTP.MethodNotAllowed { }
+
+class ValidationError extends HTTP.NotAcceptable { }
```

Comment:
This also seems like a bad fit - 406 refers specifically to content negotiation headers, and so we're likely to misuse it: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

Again, 400 or 422 would be preferable I think

Reply:
I went with 400. A validation error seems more the client's responsibility...